applications/openshift/authentication/oauth_inactivity_timeout/rule.yml

```diff
@@ -0,0 +1,67 @@
+prodtype: ocp4
+
+title: "Configure OAuth server so that tokens expire after a set period of inactivity"
+
+description: |-
+    <p>
+    You can configure OAuth tokens to expire after a set period of
+    inactivity. By default, no token inactivity timeout is set.
+    </p>
+    <p>
+    The inactivity timeout can be either set in the OAuth server
+    configuration or in any of the OAuth clients. The client settings
+    override the OAuth server setting.
+    </p>
+    <p>
+    To set the OAuth server inactivity timeout, edit the OAuth server
+    object: <tt>oc edit oauth cluster</tt>
+    and set the <tt>.spec.tokenConfig.accessTokenInactivityTimeout</tt>
+    parameter to the desired value:
+    <pre>
+    apiVersion: config.openshift.io/v1
+    kind: OAuth
+    metadata:
+    ...
+    spec:
+       tokenConfig:
+         accessTokenInactivityTimeout: 400s 
+    </pre>
+    For more information on configuring the OAuth server, consult the
+    OpenShift documentation:
+    {{{ weblink(link="https://docs.openshift.com/container-platform/4.7/authentication/configuring-oauth-clients.html") }}}
+    </p>
+
+rationale: |-
+    Terminating an idle session within a short time period reduces the window
+    of opportunity for unauthorized personnel to take control of a session
+    that has been left unattended.
+
+references:
+  nist: AC-2(5),SC-10
+
+identifiers:
+    cce@ocp4: CCE-83511-6
+
+ocil_clause: 'OAuth server inactivity timeout is not configured'
+
+ocil: |-
+  To check if the OAuth server timeout is configured, run the following command:
+  <pre>oc get oauth cluster -ojsonpath='{.spec.tokenConfig.accessTokenInactivityTimeout}'</pre>
+  the output should return a timeout value.
+
+severity: medium
+
+warnings:
+    - general: |-
+        {{{ openshift_cluster_setting("/apis/config.openshift.io/v1/oauths/cluster") | indent(8) }}}
+
+template:
+    name: yamlfile_value
+    vars:
+        ocp_data: 'true'
+        filepath: /apis/config.openshift.io/v1/oauths/cluster
+        yamlpath: ".spec.tokenConfig.accessTokenInactivityTimeout"
+        check_existence: "only_one_exists"
+        values:
+            - value: ".*"
```

Comment:
At the moment, the rule checks for ANY timeout. I was going back and forth on whether we should already introduce a variable and check for its value to make sure the timeout is not higher than some organizational setting. But 1) I guess the timeouts can be different for every client which would make the check difficult and 2) since the remediation is static anyway, setting the variable to a custom value might make the check fail because the remediation would set it to something else than the check looks for.
Thoughts?

Reply:
I'm fine with this, but ultimately it should be a variable. If you plan to leave this as it is, let's add a TODO comment.

Reply:
Hmm, the more I think about it the more I think the variable would be better. When we are able to template the remediations, it would be easier to see which must be templated if we just look for the existing variables. I'll change the PR.